### PR TITLE
`getMetaData_v3`

### PR DESCRIPTION
**Pull request by agnxsh**
implements https://github.com/ethereum/consensus-specs/blob/dev/specs/_features/eip7594/p2p-interface.md#metadata

conditionally calls getMetaData v2 or v3 by checking if `FULU_FORK_EPOCH` is scheduled or not.

### CODE REVIEW

File: beacon_chain/networking/eth2_network.nim [hunk 1]
```diff
@@ -2082,12 +2082,32 @@ proc p2pProtocolBackendImpl*(p: P2PProtocol): Backend =
 import ./peer_protocol
 export peer_protocol
 
+proc updateMetadataV2ToV3(metadata: altair.MetaData): fulu.MetaData =
```
Comment (tersec) [inline beacon_chain/networking/eth2_network.nim]: `func`

File: beacon_chain/networking/eth2_network.nim [hunk 2]
```diff
@@ -2082,12 +2082,32 @@ proc p2pProtocolBackendImpl*(p: P2PProtocol): Backend =
...
+    seq_number: metadata.seq_number,
+    attnets: metadata.attnets,
+    syncnets: metadata.syncnets)
+
+proc getMetadata_vx(node: Eth2Node, peer: Peer): 
+                    Future[NetRes[fulu.MetaData]]
+                   {.async: (raises: [CancelledError]).} =
+  let
+    res = 
+      if node.cfg.FULU_FORK_EPOCH != FAR_FUTURE_EPOCH:
```
Comment (tersec) [inline beacon_chain/networking/eth2_network.nim:2100]: I guess it's likely/certain enough that PeerDAS or something similar will show up in Fulu that this isn't much of a gamble, so, sure.

File: beacon_chain/networking/eth2_network.nim [hunk 3]
```diff
@@ -2082,12 +2082,32 @@ proc p2pProtocolBackendImpl*(p: P2PProtocol): Backend =
...
+        # Directly fetch fulu metadata if available
+        await getMetadata_v3(peer)
+      else:
+        let metadata_v2_result = await getMetadata_v2(peer)
+        metadata_v2_result.map(proc (altairData: altair.MetaData):
```
Comment (tersec) [inline beacon_chain/networking/eth2_network.nim]: [mapIt](https://nim-lang.org/docs/sequtils.html#mapIt.t%2Ctyped%2Cuntyped) works here more simply and efficiently:
```nim
       metadata_v2_result.mapIt(it.updateMetadataV2ToV3)
```
or
```nim
       metadata_v2_result.mapIt(updateMetadataV2ToV3(it))
```

File: beacon_chain/networking/eth2_network.nim [hunk 4]
```diff
@@ -2082,12 +2082,32 @@ proc p2pProtocolBackendImpl*(p: P2PProtocol): Backend =
 import ./peer_protocol
 export peer_protocol
 
+proc updateMetadataV2ToV3(metadata: altair.MetaData): fulu.MetaData =
+  fulu.MetaData(
+    seq_number: metadata.seq_number,
+    attnets: metadata.attnets,
+    syncnets: metadata.syncnets)
+
+proc getMetadata_vx(node: Eth2Node, peer: Peer):
```
Comment (tersec) [inline beacon_chain/networking/eth2_network.nim:2095]: The name emphasizes that it works with any of the `getMetadata` versions, but at this point does that convey useful information, i.e. are there similar/comparable functions which don't have that property?

File: beacon_chain/networking/eth2_network.nim (hunk 1 above)
Comment (agnxsh) [reply beacon_chain/networking/eth2_network.nim]: ok

File: beacon_chain/networking/eth2_network.nim (hunk 4 above)
Comment (agnxsh) [reply beacon_chain/networking/eth2_network.nim:2095]: i think vx does convey the fact that it internally handles the metadata versions (whatever version there is, and is NOT deprecated).

File: beacon_chain/networking/eth2_network.nim (hunk 3 above)
Comment (agnxsh) [reply beacon_chain/networking/eth2_network.nim]: changed to a simpler/cleaner alternative

File: beacon_chain/networking/eth2_network.nim [hunk 5]
```diff
@@ -2082,12 +2082,33 @@ proc p2pProtocolBackendImpl*(p: P2PProtocol): Backend =
 import ./peer_protocol
 export peer_protocol
 
+proc updateMetadataV2ToV3(metadataRes: NetRes[altair.MetaData]): 
+                          NetRes[fulu.MetaData] =
+  if metadataRes.isOk:
+    let metadata = metadataRes.get
+    return ok(fulu.MetaData(seq_number: metadata.seq_number,
```
Comment (tersec) [inline beacon_chain/networking/eth2_network.nim]: Neither branch needs the `return`, can use implicit return style here

File: beacon_chain/networking/eth2_network.nim (hunk 1 above)
Comment (tersec) [reply beacon_chain/networking/eth2_network.nim]: not relevant anymore, other changes

**Pull request merged**